jobs/integr8ly/ocp3/nightly/osd-testing-nightly-trigger.yaml

```diff
@@ -0,0 +1,206 @@
+                                [$class: 'StringBinding', credentialsId: "rhmi-qe1-token", variable: 'OPENSHIFT_TOKEN']
+                            ]) {
+                            installationStatus = build(job: 'osd-cluster-integreatly-install', propagate: false, parameters: [
+                                string(name: 'clusterName', value: "${CLUSTER_NAME}"),
+                                string(name: 'installationGitBranch', value: "${INSTALLATION_BRANCH}"),
+                                string(name: 'routerShard', value: "${ROUTER_SHARD}"),
+                                string(name: 'openShiftToken', value: "${OPENSHIFT_TOKEN}"),
+                                string(name: 'githubClientId', value: "${GH_CLIENT_ID}"),
+                                string(name: 'githubClientSecret', value: "${GH_CLIENT_SECRET}"),
+                                string(name: 'customerAdminPassword', value: "${CUSTOMER_ADMIN_PASSWORD}"),
+                                string(name: 'threeScaleAwsAccessKey', value: "${AWS_ACCESS_KEY_ID}"),
+                                string(name: 'threeScaleAwsSecretKey', value: "${AWS_SECRET_ACCESS_KEY}"),
+                                string(name: 'threeScaleBucketName', value: "${THREESCALE_BUCKET_NAME}"),
+                                string(name: 'threeScaleBucketRegion', value: "${THREESCALE_BUCKET_REGION}"),
+                                string(name: 'backupAwsKey', value: "${AWS_ACCESS_KEY_ID}"),
+                                string(name: 'backupAwsSecret', value: "${AWS_SECRET_ACCESS_KEY}"),
+                                string(name: 'backupsBucketName', value: "${BACKUPS_BUCKET_NAME}"),
+                                string(name: 'clientSecret', value: "${SSO_SECRET}")
+                            ]).result
+                        }
+                    } // stage
+
+                    if (installationStatus != 'SUCCESS' || uninstallationStatus != 'SUCCESS') {
+                        println 'There were some errors in previous install/uninstall pipelines!'
+                        currentBuild.result = 'FAILURE'
+                        return
+                    }
+
+                    // This relies on a cronjob (or something else) to update the client secret in master-config.yaml so that we can successfully login afterwards
+                    stage ('Try to login to installed cluster') {
+                        println "Waiting for successful login to the ${CLUSTER_NAME} cluster"
+                        i = 0
+                        // wait max 1 hour
+                        while(!loggedIn && i < 360) {
+                            loggedIn = ocLogin()
+                            i += 1
+                            sleep time: 5, unit: 'MINUTES'
+                        }
+                    }
+
+                    // we want to run test only when there was a successful login to the cluster
+                    if (loggedIn) {
```

Comment:
what about having else brach making the build unstable (or failed might be even better). And outputting to log that the tests were skipped?

Reply:
yep good point